### DISCUSSION TITLE
question about conv for mu and sigma in autoencoderkl

### DISCUSSION THREAD

**Discussion by Can-Zhao** [Project-MONAI/GenerativeModels#400] [Q&A]
Hi,

In AutoencoderKL, the mu and sigma are results of convolution from h, https://github.com/Project-MONAI/GenerativeModels/blob/b1c1a459c98a4d55570fe775fdcfad1707dae71b/generative/networks/nets/autoencoderkl.py#L701-L702.

If h has only one latent channel, then this single channel value needs to have the ability to generate both mu and sigma. It seems difficult intuitively?
I was wondering if you could help me with it. @ericspod  Or should I set latent channel to be >1 ?

Thank you!

Best,
Can

**Comment by ericspod** [comment]
I would guess all that matters is how large `h` is in total, even with 1 channel it's spatial dimensions might be large enough that it has sufficient capacity to represent a compressed form of the input. I would assume this would be like any other VAE in that if your results after encoding/decoding are poor then your capacity is too small and so adding channels could help with that. @Warvito might have more insight on this.

**Reply by Can-Zhao** [reply]
Thank you so much Eric! Yes, if h is large enough, then channel=1 is also fine
